components/Ink/src/MDCInkView.m

```diff
@@ -58,14 +59,28 @@ - (void)commonMDCInkViewInit {
   self.autoresizingMask = UIViewAutoresizingFlexibleHeight | UIViewAutoresizingFlexibleWidth;
   self.inkColor = self.defaultInkColor;
   _usesLegacyInkRipple = YES;
+
+  // Create a mask layer to contain MDCInkLayer when the superview has a shadowPath.
+  _maskLayer = [CAShapeLayer layer];
+  _maskLayer.masksToBounds = YES;
+  [self.layer addSublayer:_maskLayer];
 }
 
 - (void)layoutSubviews {
   [super layoutSubviews];
+
+  // If the superview has a shadowPath make sure ink does not spread outside of the shadowPath.
+  if (self.superview.layer.shadowPath) {
+    CAShapeLayer *maskLayer = [CAShapeLayer layer];
```

Comment:
Reuse the existing maskLayer instead of creating a new CAShapeLayer each layout. If you want to animate the mask you'll need to change the path on the same layer object. See #2523 for how I animated shadowPaths and masks on MDCShadowLayer.